### PR TITLE
Update gradle version to latest

### LINKED ISSUE #2961: Update gradle version and dependent components
#### Problem description
We should bump up the current gradle version to the latest. This change will break some of the existing gradle tools like PMD due to the changes in the latest gradle configuration. So, those must be fixed as well. Apart from that, we should start exploring new features (like R8) released.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #2961 

#### What has been done to verify that this works as intended?
The project builds and gradle tasks are working as usual. Also tried running the app on a device and it works as expected.

#### Why is this the best possible solution? Were any other approaches considered?
Should be updated to make use of the latest features released which improve build performance.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Shouldn't affect users

#### Do we need any specific form for testing your changes? If so, please attach one.
Not needed

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: config/quality.gradle [hunk 1]
```diff
@@ -34,42 +34,40 @@ apply plugin: 'com.github.spotbugs'
 
 spotbugs.toolVersion = '3.1.3'
 
-android.applicationVariants.all { variant ->
```
Comment (huangyz0918) [inline config/quality.gradle:37]: For line 37 to 72, you just improved the code format? It's good, but I think unnecessary changes may confuse reviewers, please note in the descriptions if necessary.

File: collect_app/src/androidTest/java/org/odk/collect/android/AllWidgetsFormTest.java [hunk 2]
```diff
@@ -1104,7 +1104,7 @@ private String randomIntegerSeparator() {
     //endregion
 
     //region Custom TestRule.
-    private class FormEntryActivityTestRule extends IntentsTestRule<FormEntryActivity> {
+    private static class FormEntryActivityTestRule extends IntentsTestRule<FormEntryActivity> {
```
Comment (huangyz0918) [inline collect_app/src/androidTest/java/org/odk/collect/android/AllWidgetsFormTest.java:1107]: Is those changes in Tests related to the update of gradle version?

File: config/quality.gradle (hunk 1 above)
Comment (shobhitagarwal1612) [reply config/quality.gradle:37]: I have written the description in the commit

File: collect_app/src/androidTest/java/org/odk/collect/android/AllWidgetsFormTest.java (hunk 2 above)
Comment (shobhitagarwal1612) [reply collect_app/src/androidTest/java/org/odk/collect/android/AllWidgetsFormTest.java:1107]: The build was failing because of these spotbugs detections. It is possible that these are linked to the updated gradle version

**Review by huangyz0918 — APPROVED** [review]

**Review by lognaturel — APPROVED** [review]
This looks great to me with good commit messages. I'd recommend waiting until 1.21 is out, though, so it's one less change to worry about. Even if it's not user-facing, there's always some risk that things change in unpredictable ways.

**Comment by lognaturel** [comment]
@shobhitagarwal1612 I just noticed the tests are failing. Is that related to the change here or does re-running them maybe work?

**Comment by shobhitagarwal1612** [comment]
I ran it locally before pushing and it was fine. Let me check what's happening on CircleCI

**Comment by shobhitagarwal1612** [comment]
Remaining warnings are caused by the current crashlytics plugin. 
Verified by running with `-Pandroid.debug.obsoleteApi=true` flag

https://github.com/firebase/firebase-android-sdk/issues/198

File: config/quality.gradle [hunk 3]
```diff
@@ -34,8 +36,7 @@ apply plugin: 'com.github.spotbugs'
 
 spotbugs.toolVersion = '3.1.3'
 
-task("spotbugs", type: com.github.spotbugs.SpotBugsTask, dependsOn: "assembleDebug") {
-
+tasks.register("spotbugs", SpotBugsTask) {
```
Comment (grzesiek2010) [inline config/quality.gradle:39]: In one of the previous commits, you added `dependsOn: "assembleDebug"` and here you removed it, why?

Comment (shobhitagarwal1612) [reply config/quality.gradle:39]: It was redundant. Even after removing the `dependsOn: "assembleDebug"`, the task `assembleDebug` gets executed (probably some internal dependency) before running the `spotbugs` task. Verified by cleaning the build dir and then running the `spotbugs` task

**Review by grzesiek2010 — APPROVED** [review]

**Comment by grzesiek2010** [comment]
@lognaturel any concerns or we can merge?

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
No concerns now that the tests pass!
